src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/alignment/ContigAlignmentsModifier.java

```diff
@@ -39,17 +40,30 @@ public static AlignmentInterval clipAlignmentInterval(final AlignmentInterval in
         Utils.validateArg(clipLengthOnRead < input.endInAssembledContig - input.startInAssembledContig + 1,
                             "input alignment to be clipped away: " + input.toPackedString() + "\twith clip length: " + clipLengthOnRead);
 
-        final Tuple2<SimpleInterval, Cigar> result = computeNewRefSpanAndCigar(input, clipLengthOnRead, clipFrom3PrimeEnd);
+        final Tuple2<SimpleInterval, Cigar> newRefSpanAndCigar = computeNewRefSpanAndCigar(input, clipLengthOnRead, clipFrom3PrimeEnd);
+        final Tuple2<Integer, Integer> newContigStartAndEnd =
+                computeNewReadSpan(input.startInAssembledContig, input.endInAssembledContig, newRefSpanAndCigar._2,
+                        clipLengthOnRead, clipFrom3PrimeEnd);
+        return new AlignmentInterval(newRefSpanAndCigar._1, newContigStartAndEnd._1, newContigStartAndEnd._2, newRefSpanAndCigar._2,
+                input.forwardStrand, input.mapQual, AlignmentInterval.NO_NM, AlignmentInterval.NO_AS, AlnModType.UNDERGONE_OVERLAP_REMOVAL);
+    }
+
+    private static Tuple2<Integer, Integer> computeNewReadSpan(final int originalContigStart, final int originalContigEnd,
+                                                               final Cigar newCigarAlong5to3DirectionOfContig,
+                                                               final int clipLengthOnRead, final boolean clipFrom3PrimeEnd) {
         final int newTigStart, newTigEnd;
         if (clipFrom3PrimeEnd) {
-            newTigStart = input.startInAssembledContig;
-            newTigEnd   = input.endInAssembledContig - clipLengthOnRead;
+            newTigStart = originalContigStart;
+            newTigEnd   = Math.min(originalContigEnd - clipLengthOnRead,
```

Comment:
As discussed in person maybe add a couple of notes in comments to highlight why this is necessary?